test/functorch/test_eager_transforms.py

```diff
@@ -146,6 +147,29 @@ def test_argnums_reorders(self):
         res = _slice_argnums(args, (1, 0))
         self.assertEqual(res, (args[1], args[0]))
 
+def _get_weights_and_functional_call(net, mechanism):
+    if mechanism == "make_functional":
+        return make_functional(net)
+    else:
+        # this makes it so the function from make_functional and this call have the same signature
+        def net_func(weights, data):
+            return functional_call(net, weights, (data,))
+
+        weights = {k: v for k, v in net.named_parameters()}
+        return net_func, weights
+
+def _get_weights_and_functional_call_with_buffers(net, mechanism):
+    if mechanism == "make_functional":
+        return make_functional_with_buffers(net)
+    else:
+        # this makes it so the function from make_functional and this call have the same signature
+        def net_func(weights, buffers, data):
+            return functional_call(net, {**weights, **buffers}, (data,))
```

Comment:
should use functional_call(net, (weights, buffers), (data,)) to test that path of function_call